lib/at_host/at_host.c

```diff
@@ -70,49 +70,44 @@ static inline void write_uart_string(const char *str)
 	}
 }
 
-static void response_handler(void *context, const char *response)
+static void response_handler(const char *response)
 {
-	ARG_UNUSED(context);
-
 	/* Forward the data over UART */
 	write_uart_string(response);
 }
 
 static void cmd_send(struct k_work *work)
 {
 	char              str[25];
-	enum at_cmd_state state;
 	int               err;
 
 	ARG_UNUSED(work);
 
-	err = at_cmd_write(at_buf, at_buf,
-			   sizeof(at_buf), &state);
+	err = nrf_modem_at_cmd(at_buf, sizeof(at_buf), "%s", at_buf);
 	if (err < 0) {
 		LOG_ERR("Error while processing AT command: %d", err);
-		state = AT_CMD_ERROR;
 	}
 
 	/* Handle the various error responses from modem */
-	switch (state) {
-	case AT_CMD_OK:
+	switch (nrf_modem_at_err_type(err)) {
+	case 0:
 		write_uart_string(at_buf);
-		write_uart_string(OK_STR);
 		break;
-	case AT_CMD_ERROR:
+	case NRF_MODEM_AT_ERROR:
 		write_uart_string(ERROR_STR);
 		break;
-	case AT_CMD_ERROR_CMS:
+	case NRF_MODEM_AT_CMS_ERROR:
 		sprintf(str, "+CMS ERROR: %d\r\n", err);
 		write_uart_string(str);
 		break;
-	case AT_CMD_ERROR_CME:
+	case NRF_MODEM_AT_CME_ERROR:
 		sprintf(str, "+CME ERROR: %d\r\n", err);
 		write_uart_string(str);
 		break;
```

Comment:
These cases could potentially be simplified by just printing `at_buf` since at_host just prints error information and doesn't take actions based on error codes.
This is what I decided to do in modem_shell in the end after having similar code than here at first:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf9160/modem_shell/src/at/at_shell.c#L61-L70

Reply:
@trantanen As far as I understand, system tests expect this kind of string outputs and they do pattern matching on them.

Reply:
This is the output you get if you just print the response buffer from the modem, which is why I decided to just print it in modem_shell. But I'll leave it up to you to decide whether to modify it or not.